### PR TITLE
[MIRROR] Virtual Limbsanity [MDB IGNORE]

### PR DESCRIPTION
**Pull request by SkyratBot**
Original PR: https://github.com/tgstation/tgstation/pull/69841
--------------------

https://www.youtube.com/watch?v=4JkIs37a2JE

## About The Pull Request
Fixes #66251

General code cleanup, adds a new footgun I pray people don't misuse. Oh, and adds a new unit test to ensure all bodypart appearances are valid by default.

## Changelog

:cl:
fix: Robotic limbs are once again style-able using a spraycan.
/:cl:

### CODE REVIEW

**Comment by Zonespace27** [comment]
FUUUUUUUUUUU-

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -136,8 +136,11 @@ GLOBAL_LIST_EMPTY(features_by_species)
 	var/punchstunthreshold = 10 //yes it should be to the attacked race but it's not useful that way even if it's logical
 	///Base electrocution coefficient.  Basically a multiplier for damage from electrocutions.
 	var/siemens_coeff = 1
+<<<<<<< HEAD
 	///What kind of damage overlays (if any) appear on our species when wounded? If this is "", does not add an overlay.
 	var/damage_overlay_type = SPECIES_HUMAN
+=======
+>>>>>>> ab6dcbf4cab (Virtual Limbsanity (#69841))
```
Comment (tf-4) [inline code/modules/mob/living/carbon/human/species.dm]:
```suggestion
```

File: code/modules/mob/living/carbon/human/species_types/ethereal.dm [hunk 2]
```diff
@@ -14,8 +14,12 @@
 	brutemod = 1.25 //They're weak to punches
 	payday_modifier = 0.75
 	attack_type = BURN //burn bish
+<<<<<<< HEAD
 	damage_overlay_type = "" //We are too cool for regular damage overlays
 	species_traits = list(DYNCOLORS, AGENDER, HAIR, FACEHAIR, HAS_FLESH, HAS_BONE) // i mean i guess they have blood so they can have wounds too //SKYRAT EDIT - Removes Lack of Underwear
+=======
+	species_traits = list(DYNCOLORS, AGENDER, NO_UNDERWEAR, HAIR, FACEHAIR, HAS_FLESH, HAS_BONE) // i mean i guess they have blood so they can have wounds too
+>>>>>>> ab6dcbf4cab (Virtual Limbsanity (#69841))
```
Comment (tf-4) [inline code/modules/mob/living/carbon/human/species_types/ethereal.dm]:
```suggestion
	species_traits = list(DYNCOLORS, AGENDER, HAIR, FACEHAIR, HAS_FLESH, HAS_BONE) // i mean i guess they have blood so they can have wounds too //SKYRAT EDIT - Removes Lack of Underwear
```

File: code/modules/surgery/bodyparts/head.dm [hunk 3]
```diff
@@ -224,11 +220,19 @@
...
 					debrain_overlay.icon_state = "debrained_alien"
 				else if(animal_origin == LARVA_BODYPART)
 					debrain_overlay.icon = 'icons/mob/animal_parts.dmi'
+=======
+				if(bodytype & BODYTYPE_ALIEN)
+					debrain_overlay.icon = 'icons/mob/species/alien/bodyparts.dmi'
+					debrain_overlay.icon_state = "debrained_alien"
+				else if(bodytype & BODYTYPE_LARVA_PLACEHOLDER)
+					debrain_overlay.icon = 'icons/mob/species/alien/bodyparts.dmi'
+>>>>>>> ab6dcbf4cab (Virtual Limbsanity (#69841))
```
Comment (tf-4) [inline code/modules/surgery/bodyparts/head.dm]:
```suggestion
				if(bodytype & BODYTYPE_ALIEN)
					debrain_overlay.icon = 'icons/mob/species/alien/bodyparts.dmi'
					debrain_overlay.icon_state = "debrained_alien"
				else if(bodytype & BODYTYPE_LARVA_PLACEHOLDER)
					debrain_overlay.icon = 'icons/mob/species/alien/bodyparts.dmi'
```

File: code/modules/surgery/bodyparts/head.dm [hunk 4]
```diff
@@ -318,23 +322,41 @@
...
+	dmg_overlay_type = SPECIES_MONKEY
+	is_dimorphic = FALSE
 
 /obj/item/bodypart/head/alien
+<<<<<<< HEAD
 	icon = 'icons/mob/animal_parts.dmi'
+=======
+	icon = 'icons/mob/species/alien/bodyparts.dmi'
+	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
+>>>>>>> ab6dcbf4cab (Virtual Limbsanity (#69841))
```
Comment (tf-4) [inline code/modules/surgery/bodyparts/head.dm]:
```suggestion
	icon = 'icons/mob/species/alien/bodyparts.dmi'
	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
```

File: code/modules/surgery/bodyparts/head.dm [hunk 5]
```diff
@@ -318,23 +322,41 @@
...
-	animal_origin = ALIEN_BODYPART
+	bodytype = BODYTYPE_HUMANOID | BODYTYPE_ALIEN | BODYTYPE_ORGANIC
 
 /obj/item/bodypart/head/larva
+<<<<<<< HEAD
 	icon = 'icons/mob/animal_parts.dmi'
+=======
+	icon = 'icons/mob/species/alien/bodyparts.dmi'
+	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
+>>>>>>> ab6dcbf4cab (Virtual Limbsanity (#69841))
```
Comment (tf-4) [inline code/modules/surgery/bodyparts/head.dm]:
```suggestion
	icon = 'icons/mob/species/alien/bodyparts.dmi'
	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
```

File: code/modules/surgery/bodyparts/parts.dm [hunk 6]
```diff
@@ -40,25 +40,43 @@
...
 	acceptable_bodytype = BODYTYPE_MONKEY
+	dmg_overlay_type = SPECIES_MONKEY
 
 /obj/item/bodypart/chest/alien
+<<<<<<< HEAD
 	icon = 'icons/mob/animal_parts.dmi'
+=======
+	icon = 'icons/mob/species/alien/bodyparts.dmi'
+	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
+>>>>>>> ab6dcbf4cab (Virtual Limbsanity (#69841))
```
Comment (tf-4) [inline code/modules/surgery/bodyparts/parts.dm]:
```suggestion
	icon = 'icons/mob/species/alien/bodyparts.dmi'
	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
```

File: code/modules/surgery/bodyparts/parts.dm [hunk 7]
```diff
@@ -40,25 +40,43 @@
...
-	animal_origin = ALIEN_BODYPART
+	acceptable_bodytype = BODYTYPE_HUMANOID
 
 /obj/item/bodypart/chest/larva
+<<<<<<< HEAD
 	icon = 'icons/mob/animal_parts.dmi'
+=======
+	icon = 'icons/mob/species/alien/bodyparts.dmi'
+	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
+>>>>>>> ab6dcbf4cab (Virtual Limbsanity (#69841))
```
Comment (tf-4) [inline code/modules/surgery/bodyparts/parts.dm]:
```suggestion
	icon = 'icons/mob/species/alien/bodyparts.dmi'
	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
```

File: code/modules/surgery/bodyparts/parts.dm [hunk 8]
```diff
@@ -148,22 +166,29 @@
...
 	px_y = -3
+	dmg_overlay_type = SPECIES_MONKEY
 
 /obj/item/bodypart/l_arm/alien
+<<<<<<< HEAD
 	icon = 'icons/mob/animal_parts.dmi'
+=======
+	icon = 'icons/mob/species/alien/bodyparts.dmi'
+	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
+>>>>>>> ab6dcbf4cab (Virtual Limbsanity (#69841))
```
Comment (tf-4) [inline code/modules/surgery/bodyparts/parts.dm]:
```suggestion
	icon = 'icons/mob/species/alien/bodyparts.dmi'
	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
```

File: code/modules/surgery/bodyparts/parts.dm [hunk 9]
```diff
@@ -251,21 +276,28 @@
...
 	px_y = -3
+	dmg_overlay_type = SPECIES_MONKEY
 
 /obj/item/bodypart/r_arm/alien
+<<<<<<< HEAD
 	icon = 'icons/mob/animal_parts.dmi'
+=======
+	icon = 'icons/mob/species/alien/bodyparts.dmi'
+	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
+>>>>>>> ab6dcbf4cab (Virtual Limbsanity (#69841))
```
Comment (tf-4) [inline code/modules/surgery/bodyparts/parts.dm]:
```suggestion
	icon = 'icons/mob/species/alien/bodyparts.dmi'
	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
```

File: code/modules/surgery/bodyparts/parts.dm [hunk 10]
```diff
@@ -344,21 +376,27 @@
...
 	px_y = 4
+	dmg_overlay_type = SPECIES_MONKEY
 
 /obj/item/bodypart/l_leg/alien
+<<<<<<< HEAD
 	icon = 'icons/mob/animal_parts.dmi'
+=======
+	icon = 'icons/mob/species/alien/bodyparts.dmi'
+	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
+>>>>>>> ab6dcbf4cab (Virtual Limbsanity (#69841))
```
Comment (tf-4) [inline code/modules/surgery/bodyparts/parts.dm]:
```suggestion
	icon = 'icons/mob/species/alien/bodyparts.dmi'
	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
```

File: code/modules/surgery/bodyparts/parts.dm [hunk 11]
```diff
@@ -440,22 +478,33 @@
 		owner.set_usable_legs(owner.usable_legs + 1)
 
 /obj/item/bodypart/r_leg/monkey
+<<<<<<< HEAD
 	icon = 'icons/mob/animal_parts.dmi'
+=======
+	icon = 'icons/mob/species/monkey/bodyparts.dmi'
+	icon_static = 'icons/mob/species/monkey/bodyparts.dmi'
+>>>>>>> ab6dcbf4cab (Virtual Limbsanity (#69841))
```
Comment (tf-4) [inline code/modules/surgery/bodyparts/parts.dm]:
```suggestion
	icon = 'icons/mob/species/monkey/bodyparts.dmi'
	icon_static = 'icons/mob/species/monkey/bodyparts.dmi'
```

File: code/modules/surgery/bodyparts/parts.dm [hunk 12]
```diff
@@ -440,22 +478,33 @@
...
 	px_y = 4
+	dmg_overlay_type = SPECIES_MONKEY
 
 /obj/item/bodypart/r_leg/alien
+<<<<<<< HEAD
 	icon = 'icons/mob/animal_parts.dmi'
+=======
+	icon = 'icons/mob/species/alien/bodyparts.dmi'
+	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
+>>>>>>> ab6dcbf4cab (Virtual Limbsanity (#69841))
```
Comment (tf-4) [inline code/modules/surgery/bodyparts/parts.dm]:
```suggestion
	icon = 'icons/mob/species/alien/bodyparts.dmi'
	icon_static = 'icons/mob/species/alien/bodyparts.dmi'
```

**Comment by GoldenAlpharex** [comment]
We're going to test-merge this tonight before actually merging it, I don't want to have to do some emergency fixing afterwards to ensure that X or Y important visual feature works again.

**Comment by GoldenAlpharex** [comment]
It worked, I'll just have to update the examine panel but I'll do that tomorrow.

**Pull request merged**